extensions/kafka/src/test/java/com/hazelcast/jet/kafka/impl/KafkaTestSupport.java

```diff
@@ -113,45 +148,74 @@ public void setPartitionCount(String topicId, int numPartitions) {
+    public Future<RecordMetadata> produce(String topic, Object key, Object value) {
+        return producers.computeIfAbsent(topic, t -> getProducer(topic, key, value))
+                .send(new ProducerRecord<>(topic, key, value));
     }
 
-    Future<RecordMetadata> produce(String topic, int partition, Long timestamp, Integer key, String value) {
-        return getProducer().send(new ProducerRecord<>(topic, partition, timestamp, key, value));
+    public Future<RecordMetadata> produce(String topic, int partition, Long timestamp, Object key, Object value) {
+        return producers.computeIfAbsent(topic, t -> getProducer(topic, key, value))
+                .send(new ProducerRecord<>(topic, partition, timestamp, key, value));
     }
 
-    private KafkaProducer<Integer, String> getProducer() {
-        if (producer == null) {
-            Properties producerProps = new Properties();
-            producerProps.setProperty("bootstrap.servers", brokerConnectionString);
-            producerProps.setProperty("key.serializer", IntegerSerializer.class.getCanonicalName());
-            producerProps.setProperty("value.serializer", StringSerializer.class.getCanonicalName());
-            producerProps.setProperty("max.block.ms", String.valueOf(KAFKA_MAX_BLOCK_MS));
-            producer = new KafkaProducer<>(producerProps);
+    private KafkaProducer<Object, Object> getProducer(String topic, Object key, Object value) {
+        Properties producerProps = new Properties();
+        producerProps.setProperty("bootstrap.servers", brokerConnectionString);
+        producerProps.setProperty("key.serializer", resolveSerializer(key));
+        producerProps.setProperty("value.serializer", resolveSerializer(value));
+        if (key instanceof GenericRecord || value instanceof GenericRecord) {
+            producerProps.setProperty("schema.registry.url", getSchemaRegistryURI().toString());
         }
-        return producer;
+        producerProps.setProperty("max.block.ms", String.valueOf(KAFKA_MAX_BLOCK_MS));
+        Optional.ofNullable(producerProperties.get(topic)).ifPresent(producerProps::putAll);
+        return new KafkaProducer<>(producerProps);
     }
 
-    private KafkaProducer<String, String> getStringStringProducer() {
-        if (stringStringProducer == null) {
-            Properties producerProps = new Properties();
-            producerProps.setProperty("bootstrap.servers", brokerConnectionString);
-            producerProps.setProperty("key.serializer", StringSerializer.class.getCanonicalName());
-            producerProps.setProperty("value.serializer", StringSerializer.class.getCanonicalName());
-            stringStringProducer = new KafkaProducer<>(producerProps);
+    /**
+     * @see org.apache.kafka.common.serialization.Serdes#serdeFrom(Class)
+     * @see com.hazelcast.jet.sql.impl.connector.kafka.PropertiesResolver#resolveSerializer(String)
```

Comment:
wouldn't it be easier to make `PropertiesResolver` resolveSerializer and resolveDeserializer methods visible to tests?

Reply:
`PropertiesResolver` only resolves types that are known to SQL. `KafkaTestSupport` is a Jet utility and should support every type that has a builtin serde.